### PR TITLE
Ft post articles  gifs

### PR DESCRIPTION
**Pull request by goodnessaig1**

### CODE REVIEW

File: src/Auth/Actions/articleActions.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    };
+};
+export function getCategory() {
+    return (dispatch) => {
+        const promise = apiRequest('GET', `v1/categories`);
+        dispatch(getCategoryRequest());
+        promise.then(
+            function (payload) {
+                const categories = payload.data;
+                if (categories.status === 'Failed') {
```
Comment (courage173) [inline src/Auth/Actions/articleActions.js]: why are you checking this status if we have a catch block where we are handling errors?

File: src/Auth/Actions/articleActions.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+export function PostArticles(credentials) {
+    return (dispatch) => {
+        const promise = apiRequest('POST', `v1/articles`, credentials);
+        dispatch(PostArticlesRequest());
+        promise.then(
+            function (payload) {
+                const article = payload.data;
+                if (article.status === 'Failed') {
```
Comment (courage173) [inline src/Auth/Actions/articleActions.js]: same

File: src/Auth/Actions/userActions.js [hunk 3]
```diff
@@ -172,11 +172,12 @@ export function LoginUser(credentials, history, setFieldError, setSubmitting) {
                     const userData = data;
                     const token = userData.data.token;
                     localStorage.setItem('token', token);
+                    dispatch(getUserDetails());
                     dispatch(loginUserSuccess(data));
                     history.push('/dashboard');
                     setSubmitting(false);
+                    window.location.reload(true);
```
Comment (courage173) [inline src/Auth/Actions/userActions.js]: why do you have this here??

File: src/Auth/Reducer/gifReducer.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+            });
+
+        case types.POST_GIF_SUCCESS:
+            return Object.assign({}, state, {
+                PostGif: {
+                    requesting: false,
+                    error: null,
+                    success: true,
+                },
+                success: action.payload,
```
Comment (courage173) [inline src/Auth/Reducer/gifReducer.js]: please why is this key success?

File: src/Components/Dashboard/MainPage.js [hunk 5]
```diff
@@ -6,19 +6,32 @@ import { getFeedDetails } from '../../Auth/Actions/feedActions';
...
+    const [data, setData] = useState();
     useEffect(() => {
         window.scrollTo({
             top: 0,
             left: 0,
             behavior: 'smooth',
         });
-        dispatch(getFeedDetails(offSet));
+        dispatch(getFeedDetails(offSet)).then((res) => {
+            setData(res.data.data);
```
Comment (courage173) [inline src/Components/Dashboard/MainPage.js]: please get feed data from the state!!

File: src/Components/Pages/SideDrawer.js [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+import {
+    Drawer,
+    List,
+    ListItem,
+    ListItemIcon,
+    ListItemText,
+} from '@material-ui/core';
+import { LogoutUser } from '../../Auth/Actions/userActions';
+
+const SideDrawer = ({ LogoutUser, userStatus }) => {
```
Comment (courage173) [inline src/Components/Pages/SideDrawer.js]: Build this side drawer with yourself without using material UI

File: src/Components/Dashboard/Feeds.js [hunk 7]
```diff
@@ -1,110 +1,100 @@
...
         <div>
             {!requesting && (
                 <div className="dash_board_container">
-                    {feeds &&
-                        feeds.map((item, index) => (
-                            <div key={index}>
-                                <div className="feed_content">
-                                    {item.post.includes('https://') ? (
+                    <InfiniteScroll
+                        dataLength={200}
```
Comment (courage173) [inline src/Components/Dashboard/Feeds.js]: the data length should be the length of what the data you are fetching which is 10

File: src/Components/Dashboard/Feeds.js [hunk 8]
```diff
@@ -1,110 +1,100 @@
...
+                    pageYOffset = window.pageYOffset;
+                    setData(feeds.concat(res.data.data));
+                });
+            }, 3000);
+        } else {
+            setHasMore(false);
+        }
+    };
+    // console.log(pageYOffset);
+    React.useLayoutEffect(() => {
```
Comment (courage173) [inline src/Components/Dashboard/Feeds.js]: why do you need to scroll?

File: src/Components/Dashboard/Feeds.js [hunk 9]
```diff
@@ -1,110 +1,100 @@
...
+    setOffSet,
+    feedsLength,
+    requesting,
+}) => {
+    let pageYOffset = window.pageYOffset;
+    // console.log(feeds);
+    const dispatch = useDispatch();
+    const [hasMore, setHasMore] = useState(true);
+    const fetchMoreData = () => {
+        if (feedsLength >= 10) {
```
Comment (courage173) [inline src/Components/Dashboard/Feeds.js]: this check should be against the total number of items in the db and the total number of items you are rendering

**Review by courage173 — CHANGES_REQUESTED** [review]

File: src/Components/Dashboard/MainPage.js (hunk 5 above)
Comment (goodnessaig1) [reply src/Components/Dashboard/MainPage.js]: Ok Sir

File: src/Components/Dashboard/Feeds.js (hunk 8 above)
Comment (goodnessaig1) [reply src/Components/Dashboard/Feeds.js]: i used it to check if i can pageOffset to the current scroll position due to the issues with loading more data with a mobile phone

File: src/Components/Dashboard/Feeds.js (hunk 7 above)
Comment (goodnessaig1) [reply src/Components/Dashboard/Feeds.js]: Ok sir

**Pull request merged**